resources/[soz]/soz-core/src/client/shop/jewelry.shop.provider.ts

```diff
@@ -0,0 +1,112 @@
+import { FemaleJewelryItems, MaleJewelryItems } from '@public/config/jewelry';
+import { ShopBrand } from '@public/config/shops';
+import { OnNuiEvent } from '@public/core/decorators/event';
+import { Inject } from '@public/core/decorators/injectable';
+import { Provider } from '@public/core/decorators/provider';
+import { NuiEvent, ServerEvent } from '@public/shared/event';
+import { MenuType } from '@public/shared/nui/menu';
+import { Vector3 } from '@public/shared/polyzone/vector';
+import { JewelryShopItem } from '@public/shared/shop';
+
+import { CameraService } from '../camera';
+import { ClothingService } from '../clothing/clothing.service';
+import { NuiMenu } from '../nui/nui.menu';
+import { PlayerService } from '../player/player.service';
+import { ResourceLoader } from '../resources/resource.loader';
+import { ClothingShopRepository } from '../resources/shop.repository';
+
+@Provider()
+export class JewelryShopProvider {
+    @Inject(ClothingShopRepository)
+    private clothingShopRepository: ClothingShopRepository;
+
+    @Inject(PlayerService)
+    private playerService: PlayerService;
+
+    @Inject(NuiMenu)
+    private nuiMenu: NuiMenu;
+
+    @Inject(CameraService)
+    private cameraService: CameraService;
+
+    @Inject(ResourceLoader)
+    private resourceLoader: ResourceLoader;
+
+    @Inject(ClothingService)
+    private clothingService: ClothingService;
+
+    public async openShop() {
+        const modelHash = GetEntityModel(PlayerPedId());
+        const shop_content = modelHash == 1885233650 ? MaleJewelryItems : FemaleJewelryItems;
+
+        await this.setupShop();
+
+        this.nuiMenu.openMenu(MenuType.JewelryShop, { shop_content });
+    }
+
+    public async setupShop() {
+        const ped = PlayerPedId();
+        SetEntityHeading(ped, 199.156);
+        FreezeEntityPosition(ped, true);
+
+        // Setup cam
+        const [x, y, z] = GetEntityCoords(ped);
+        await this.cameraService.setupCamera([x, y - 0.5, z + 0.7] as Vector3, [x, y, z + 0.6] as Vector3);
+
+        // Play idle animation
+        const animDict = 'anim@heists@heist_corona@team_idles@male_c';
+        this.resourceLoader.loadAnimationDictionary(animDict);
+
+        ClearPedTasksImmediately(ped);
+        TaskPlayAnim(ped, animDict, 'idle', 1.0, 1.0, -1, 1, 1, false, false, false);
+    }
+
+    public async clearAllAnimations() {
+        const ped = PlayerPedId();
+        ClearPedTasks(ped);
+        this.resourceLoader.unloadAnimationDictionary('anim@heists@heist_corona@team_idles@male_c');
+    }
```

Comment:
You have 3 times this function in your shops. Maybe you should create utility file to avoid this copy/paste

Reply:
Done. Function added to `ShopService`